Update Request: bash-language-server 5.4.0 → 5.4.3

### Nixpkgs version

- Unstable (25.05)

### Package name

bash-language-server

### Upstream version

5.4.3

### Nixpkgs version

5.4.0

### Changelog

https://github.com/bash-lsp/bash-language-server/releases/tag/server-5.4.3

### Notify maintainers

@doronbehar 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this package update does not yet exist in an [open pull request](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+package+%28update%29%22) or in [Nixpkgs Unstable](https://search.nixos.org/packages?channel=unstable).
- [x] I assert that this is not a [duplicate of any known issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%229.needs%3A+package+%28update%29%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

@Scrumplex maybe you could help here a bit? Upstream updated their pnpm lockfile version to `9`, and now I'm getting modules not found errors when trying to build:

```
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }

> @ compile /build/source
> tsc -b "server"

Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Packages are copied from the content-addressable store to the virtual store.
  Content-addressable store is at: /build/tmp.x4sQWXnYVr/v3
  Virtual store is at:             ../../nix/store/pm7sd3s45y53sv8w4754knnama0i96jd-bash-language-server-5.4.3/lib/bash-language-server/node_modules/.pnpm
Progress: resolved 0, reused 1, downloaded 0, added 0
.                                        |  WARN  deprecated eslint@8.57.1
Progress: resolved 0, reused 1, downloaded 0, added 0
Progress: resolved 306, reused 306, downloaded 0, added 0
 ERR_PNPM_NO_OFFLINE_META  Failed to resolve @babel/core@7.22.6 in package mirror /build/tmp.NgkJIvyajS/.cache/pnpm/metadata-v1.3/registry.npmjs.org/@babel/core.json

This error happened while installing a direct dependency of /build/source
Progress: resolved 306, reused 306, downloaded 0, added 0
```

Upstream mentions letting the Pnpm `postinstall` scripts install dependencies for subpackages, but that didn't help - I tried:

```patch
diff --git i/pkgs/by-name/ba/bash-language-server/package.nix w/pkgs/by-name/ba/bash-language-server/package.nix
index 3a0414e777f2..876c8d09f03d 100644
--- i/pkgs/by-name/ba/bash-language-server/package.nix
+++ w/pkgs/by-name/ba/bash-language-server/package.nix
@@ -21,14 +21,18 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   pnpmWorkspaces = [ "bash-language-server" ];
+  pnpmInstallFlags = [
+    "--no-ignore-scripts"
+  ];
   pnpmDeps = pnpm_9.fetchDeps {
     inherit (finalAttrs)
       pname
       version
       src
       pnpmWorkspaces
+      pnpmInstallFlags
       ;
-    hash = "sha256-lRfVM20c2OMbPk/dUiuEv07Fu6UujM42xJ5Uc9SpQa4=";
+    hash = "sha256-majJLW2OXak+ygFyk80OAytyFnEtw7OjB1/crZe+jcg=";
   };
 
   nativeBuildInputs = [
```

And with it:

```
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }

> @ compile /build/source
> tsc -b "server"

Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Packages are copied from the content-addressable store to the virtual store.
  Content-addressable store is at: /build/tmp.TQoUpzSDJn/v3
  Virtual store is at:             ../../nix/store/r6a386ds8nhsrcg8j1m7gnpdjndg91a0-bash-language-server-5.4.3/lib/bash-language-server/node_modules/.pnpm
Progress: resolved 0, reused 1, downloaded 0, added 0
.                                        |  WARN  deprecated eslint@8.57.1
Progress: resolved 0, reused 1, downloaded 0, added 0
Progress: resolved 311, reused 311, downloaded 0, added 0
 ERR_PNPM_NO_OFFLINE_META  Failed to resolve @babel/core@7.22.6 in package mirror /build/tmp.6pRtaCSf25/.cache/pnpm/metadata-v1.3/registry.npmjs.org/@babel/core.json

This error happened while installing a direct dependency of /build/source
Progress: resolved 311, reused 311, downloaded 0, added 0
```

The current build phase of this package is basically `pnpm compile server`.

--- a/pkgs/by-name/ba/bash-language-server/package.nix
+++ b/pkgs/by-name/ba/bash-language-server/package.nix
@@ -2,7 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  pnpm_8,
+  pnpm_10,
   nodejs,
   makeBinaryWrapper,
   shellcheck,
@@ -11,29 +11,29 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "bash-language-server";
-  version = "5.4.0";
+  version = "5.6.0";
 
   src = fetchFromGitHub {
     owner = "bash-lsp";
     repo = "bash-language-server";
     rev = "server-${finalAttrs.version}";
-    hash = "sha256-yJ81oGd9aNsWQMLvDSgMVVH1//Mw/SVFYFIPsJTQYzE=";
+    hash = "sha256-Pe32lQSlyWcyUbqwhfoulwNwhrnWdRcKFIl3Jj0Skac=";
   };
 
   pnpmWorkspaces = [ "bash-language-server" ];
-  pnpmDeps = pnpm_8.fetchDeps {
+  pnpmDeps = pnpm_10.fetchDeps {
     inherit (finalAttrs)
       pname
       version
       src
       pnpmWorkspaces
       ;
-    hash = "sha256-W25xehcxncBs9QgQBt17F5YHK0b+GDEmt27XzTkyYWg=";
+    hash = "sha256-NvyqPv5OKgZi3hW98Da8LhsYatmrzrPX8kLOfLr+BrI=";
   };
 
   nativeBuildInputs = [
     nodejs
-    pnpm_8.configHook
+    pnpm_10.configHook
     makeBinaryWrapper
     versionCheckHook
   ];
@@ -45,19 +45,20 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postBuild
   '';
 
+  # will be fixed in a later commit
+  dontCheckForBrokenSymlinks = true;
+
   installPhase = ''
     runHook preInstall
 
-    pnpm --offline \
-      --frozen-lockfile --ignore-script \
-      --filter=bash-language-server \
-      deploy --prod $out/lib/bash-language-server
+    mkdir -p $out/{bin,lib/bash-language-server}
+    cp -r {node_modules,server} $out/lib/bash-language-server/
 
     # Create the executable, based upon what happens in npmHooks.npmInstallHook
     makeWrapper ${lib.getExe nodejs} $out/bin/bash-language-server \
       --suffix PATH : ${lib.makeBinPath [ shellcheck ]} \
       --inherit-argv0 \
-      --add-flags $out/lib/bash-language-server/out/cli.js
+      --add-flags $out/lib/bash-language-server/server/out/cli.js
 
     runHook postInstall
   '';

--- a/pkgs/by-name/ba/bash-language-server/package.nix
+++ b/pkgs/by-name/ba/bash-language-server/package.nix
@@ -78,9 +78,11 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   meta = {
     description = "Language server for Bash";
     homepage = "https://github.com/bash-lsp/bash-language-server";
+    changelog = "https://github.com/bash-lsp/bash-language-server/releases/tag/${finalAttrs.src.tag}";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [
       doronbehar
+      gepbird
     ];
     mainProgram = "bash-language-server";
     platforms = lib.platforms.all;

--- a/pkgs/by-name/ba/bash-language-server/package.nix
+++ b/pkgs/by-name/ba/bash-language-server/package.nix
@@ -1,6 +1,6 @@
 {
   lib,
-  stdenv,
+  stdenvNoCC,
   fetchFromGitHub,
   pnpm_10,
   nodejs,
@@ -9,14 +9,14 @@
   versionCheckHook,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "bash-language-server";
   version = "5.6.0";
 
   src = fetchFromGitHub {
     owner = "bash-lsp";
     repo = "bash-language-server";
-    rev = "server-${finalAttrs.version}";
+    tag = "server-${finalAttrs.version}";
     hash = "sha256-Pe32lQSlyWcyUbqwhfoulwNwhrnWdRcKFIl3Jj0Skac=";
   };
 
@@ -75,12 +75,14 @@ stdenv.mkDerivation (finalAttrs: {
 
   doInstallCheck = true;
 
-  meta = with lib; {
+  meta = {
     description = "Language server for Bash";
     homepage = "https://github.com/bash-lsp/bash-language-server";
-    license = licenses.mit;
-    maintainers = with maintainers; [ doronbehar ];
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [
+      doronbehar
+    ];
     mainProgram = "bash-language-server";
-    platforms = platforms.all;
+    platforms = lib.platforms.all;
   };
 })

--- a/pkgs/by-name/ba/bash-language-server/package.nix
+++ b/pkgs/by-name/ba/bash-language-server/package.nix
@@ -45,8 +45,18 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postBuild
   '';
 
-  # will be fixed in a later commit
-  dontCheckForBrokenSymlinks = true;
+  preInstall = ''
+    # remove unnecessary files
+    rm node_modules/.modules.yaml
+    pnpm --ignore-scripts --prod prune
+    rm -r node_modules/.pnpm/@mixmark-io*/node_modules/@mixmark-io/domino/{test,.yarn}
+    find -type f \( -name "*.ts" -o -name "*.map" \) -exec rm -rf {} +
+    # https://github.com/pnpm/pnpm/issues/3645
+    find node_modules server/node_modules -xtype l -delete
+
+    # remove non-deterministic files
+    rm node_modules/{.modules.yaml,.pnpm-workspace-state.json}
+  '';
 
   installPhase = ''
     runHook preInstall

